src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/IncrementalGeneratorInitializationContextExtensions.cs

```diff
@@ -84,10 +84,12 @@ public static void RegisterConcatenatedSyntaxOutputs<TNode>(this IncrementalGene
 
                     StringBuilder source = new();
                     // Mark in source that the file is auto-generated.
-                    source.AppendLine("// <auto-generated/>");
+                    // Explicitly unify on "\r\n" line endings to avoid issues with different line endings in the generated sources.
+                    source.Append("// <auto-generated/>\r\n");
```

Comment:
I'm assume this is what the other source generators do? I assume we can all agree whatever one does, they are should do with respect to line endings.

Reply:
Other generators generate strings with the system line endings. However, since we use Roslyn, we get their line endings support (which defaults to CRLF). And we can't use Environment.NewLine as Roslyn bans the Environment class in analyzers.

We're only hitting this now because we're trying to use the Roslyn testing SDK's support for comparing against known output, which is line-ending sensitive.